lib/skate/open_route_service_api/directions_request.ex

```diff
@@ -4,13 +4,46 @@ defmodule Skate.OpenRouteServiceAPI.DirectionsRequest do
   """
   @derive Jason.Encoder
 
+  defmodule Options do
+    @moduledoc false
+    defmodule ProfileParams do
+      @moduledoc false
+      defmodule HgvRestrictions do
+        @moduledoc false
+        @type t :: %{length: float(), width: float(), height: float()}
+        defstruct [:length, :width, :height]
+
+        def bus_40ft,
+          do: %{
+            length: 12.192,
+            width: 3.2004,
+            height: 3.5052
+          }
+      end
+
+      @type t :: %{restrictions: HgvRestrictions.t()}
+      defstruct [:restrictions]
+    end
+
+    @type t :: %{profile_params: ProfileParams.t()}
+    defstruct [:profile_params]
+  end
+
   @typedoc """
     Type that represents a request made to OpenRouteService's Directions API
   """
   @type t() :: %__MODULE__{
           coordinates: [[float()]],
-          continue_straight: boolean()
+          continue_straight: boolean(),
+          options: Options.t()
         }
 
-  defstruct coordinates: [], continue_straight: true
+  defstruct coordinates: [],
+            continue_straight: true,
+            options: %{
+              profile_params: %{
+                restrictions:
+                  Skate.OpenRouteServiceAPI.DirectionsRequest.Options.ProfileParams.HgvRestrictions.bus_40ft()
```

Comment:
this name is so long lol
1. Should we move the `OpenRouteServiceAPI` to the root? so there's no `Skate.` prefix anymore?
2. should we alias this so this call isn't so long?